velox/functions/sparksql/Arithmetic.h

```diff
@@ -40,6 +40,8 @@ struct RemainderFunction {
     // int64_t.
     if (UNLIKELY(n == 1 || n == -1)) {
       result = 0;
+    } else if constexpr (std::is_same_v<TInput, float> || std::is_same_v<TInput, double>) {
+      result = std::fmod(a, n);
```

Comment:
When a or n are some special values like NaN, Inf, -Inf, 0.0, -0.0, I wonder if this implementation yields consistent results with Spark. Would you help check? We may also need to document the edge cases.

Reply:
Sorry for the delayed response. I compared the results with Spark's behavior and made the necessary fixes. Here's a summary of how the `mod` function handles special values now:
| `a`                  | `n`                  | sparkResult  | velox Result       | Notes                                                                 |
|----------------------|----------------------|------------------|---------------------|-----------------------------------------------------------------------|
| `NaN`                | `1.0`                | `NaN`            | `NaN`               | Correctly returns `NaN`                                               |
| `NaN`                | `NaN`                | `NaN`            | `NaN`               | Correctly returns `NaN`                                               |
| `Infinity`           | `1.0`                | `NaN`            | `NaN`               | Correctly returns `NaN`                                               |
| `-Infinity`          | `1.0`                | `NaN`            | `NaN`               | Correctly returns `NaN`                                               |
| `Infinity`           | `Infinity`           | `NaN`            | `NaN`               | Correctly returns `NaN`                                               |
| `Infinity`           | `0.0`                | `nullopt`                | `nullopt`           | Correctly handles division by zero                           |
| `-Infinity`          | `0.0`                | `nullopt`                | `nullopt`           | Correctly handles division by zero 
| `2.14159`            | `0.0`                | `nullopt`        | `nullopt`           | Correctly handles division by zero                            |
| `Infinity`           | `-Infinity`          | `NaN`            | `NaN`               | Correctly returns `NaN`                                               |
| `NaN`                | `-Infinity`          | `NaN`            | `NaN`               | Correctly returns `NaN`                                               |
| `0.0`                | `1.0`                | `0.0`            | `0.0`               | Correctly computes `0.0 % 1.0`                                         |
| `-0.0`               | `1.0`                | `-0.0`           | `-0.0`              | Correctly computes `-0.0 % 1.0`                                        |
| `2.14159`            | `-0.0`               | `nullopt`        | `nullopt`           | Correctly handles division by zero with `-0.0`                         |

Reply:
Thanks for your efforts on validating and supporting these special values.